lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -2040,28 +2039,36 @@ instance DecodeAddress ('Testnet pm) where
 
 decodeBytes :: Text -> Either TextDecodingError ByteString
 decodeBytes t =
-    case tryBase16 t <|> tryBech32 t <|> tryBase58 t of
+    case tryBech32 t <|> tryBase58 t of
         Just bytes ->
             Right bytes
         _ ->
-            Left $ TextDecodingError
-                "Unrecognized address encoding: must be either bech32, base58 or base16"
+            Left $ TextDecodingError $ mconcat
+                [ "Unrecognized address encoding: must be either bech32 or base58."
+                , " Perhaps your address is not entirely right?"
+                , " Every character has to be correct!"
+                ]
 
--- | Attempt decoding an 'Address' using a Bech32 encoding.
+-- | Attempt to decode a Shelley 'Address' using a Bech32 encoding.
 tryBech32 :: Text -> Maybe ByteString
-tryBech32 text = do
-    (_, dp) <- either (const Nothing) Just (Bech32.decodeLenient text)
-    dataPartToBytes dp
+tryBech32 = fmap CA.unAddress . CA.fromBech32
```

Comment:
By following `CA.fromBech32` to its definition, I think I have confirmed that `fromBech32` does indeed also use `Bech32.decodeLenient`. However, I just want to check: do we have regression test coverage to warn us if this ever changes?

(I ask, because it's just inside the realm of possibility that someone might change the definition of `CA.fromBech32` to use `Bech32.decode`, rather than its more lenient variant.)

Reply:
Hm. I'm not sure if we should check this, for two reasons:

1. The address format is standardized in [CIP-0019][1] which lists the `cardano-addresses` package as a reference implementation. So, regardless of how `Cardano.Address.fromBech32` is implemented, it must be correct — by virtue of being a reference. 😅
2. Using `Bech32.decode` instead of `Bech32.decodeLenient` would actually not be a regression — as far as the definition and error correcting properties of the Bech32 encoding are concerned. 🙈  ([BIP-0173][bech32]: "A Bech32 string is at most 90 characters long and consists of […]"). Shelley addresses are typically 57 bytes in length (1 byte header + 28 bytes payment part + 28 bytes delegation part), so that would work out. The big exception are pointer addresses which can be arbitrarily long, but of which there are only a handful of instances on mainnet. 🙈 

  [1]: https://github.com/cardano-foundation/CIPs/tree/master/CIP-0019#reference-implementations
  [bech32]: https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki

Reply:
Oh wait. Even the regular Shelley addresses would not quite fit into the 90 character limit, as 57 bytes = 456 bits = 92.5 * (5 bits), and one character encodes 5 bits.